sycl/plugins/unified_runtime/ur/adapters/hip/context.hpp

```diff
@@ -1,144 +0,0 @@
-//===--------- context.hpp - HIP Adapter ----------------------------===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===-----------------------------------------------------------------===//
-#pragma once
-
-#include "common.hpp"
-#include "device.hpp"
-#include "platform.hpp"
-
-typedef void (*ur_context_extended_deleter_t)(void *UserData);
-
-/// UR context mapping to a HIP context object.
-///
-/// There is no direct mapping between a HIP context and a UR context.
-/// The main differences are described below:
-///
-/// <b> HIP context vs UR context </b>
-///
-/// One of the main differences between the UR API and the HIP driver API is
-/// that the second modifies the state of the threads by assigning
-/// `hipCtx_t` objects to threads. `hipCtx_t` objects store data associated
-/// with a given device and control access to said device from the user side.
-/// UR API context are objects that are passed to functions, and not bound
-/// to threads.
-/// The ur_context_handle_t_ object doesn't implement this behavior. It only
-/// holds the HIP context data. The RAII object \ref ScopedContext implements
-/// the active context behavior.
-///
-/// <b> Primary vs UserDefined context </b>
-///
-/// HIP has two different types of context, the Primary context,
-/// which is usable by all threads on a given process for a given device, and
-/// the aforementioned custom contexts.
-/// The HIP documentation, and performance analysis, suggest using the Primary
-/// context whenever possible. The Primary context is also used by the HIP
-/// Runtime API. For UR applications to interop with HIP Runtime API, they have
-/// to use the primary context - and make that active in the thread. The
-/// `ur_context_handle_t_` object can be constructed with a `kind` parameter
-/// that allows to construct a Primary or `UserDefined` context, so that
-/// the UR object interface is always the same.
-///
-///  <b> Destructor callback </b>
-///
-///  Required to implement CP023, SYCL Extended Context Destruction,
-///  the UR Context can store a number of callback functions that will be
-///  called upon destruction of the UR Context.
-///  See proposal for details.
-///  https://github.com/codeplaysoftware/standards-proposals/blob/master/extended-context-destruction/index.md
-///
-struct ur_context_handle_t_ {
```

Comment:
Why do we need to move this object also to `device.hpp`?

Reply:
because Hugh asked me to

Reply:
https://github.com/intel/llvm/pull/10672#discussion_r1283023499

Reply:
Personally, I still feel `ur_context_handle_t_` is more of a context thing not a device thing even from its naming, but feel free to do what you see fit.

Reply:
Oh good grief. I didn't even realise I'd moved more than one thing. Good catch. I'll semi-revert my stupidity

Reply:
Sorry yes I was referring just to the `ScopedDevice` class

Reply:
Right. Sorted